staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js

```diff
@@ -0,0 +1,413 @@
+    }else{
+        document.getElementById("welcome").value += "Nombre válido mínimo 2 carácteres"
+        document.getElementById("nameUser").value = ""
+    }
+    
+}
+
+
+n = 0
+function start(){
+    document.getElementById('letters').style="display:block;"
+    document.getElementById('showQuestion').style="display:block;"
+    document.getElementById('userAnswer').style="display:block;"
+    document.getElementById('loginUser').style="display:none;" 
+    document.getElementById('acceptar').style="display:block;"
+    document.getElementById('end').style="display:block;"
+    document.getElementById("time").style="display:block;"
+    document.getElementById("tiempo").style="display:block;"
+    document.getElementById("minuts").style="display:block;"
+    timer();
+    document.getElementById("showQuestion").value = questions[n].question;
+}
+
+// timer 
+
+function timer(){
+    startTimer()
+}
+
+function startTimer(duration, display) {
+    var timer = duration, minutes, seconds;
+    var __timer = setInterval(function () {
+        minutes = parseInt(timer / 60, 10)
+        seconds = parseInt(timer % 60, 10);
+
+        minutes = minutes < 10 ? "0" + minutes : minutes;
+        seconds = seconds < 10 ? "0" + seconds : seconds;
+
+        display.textContent = minutes + ":" + seconds;
+
+        if (--timer < 0) {
+            setInterval(__timer);
+            timer = duration;
+            document.getElementById("timeFinished").style="display:block;"
+            document.getElementById("timeFinished").value = "Tiempo acabado"
+            document.getElementById("time").style="display:none;"
+            document.getElementById("tiempo").style="display:none;"
+            document.getElementById("minuts").style="display:none;"
+            document.getElementById('letters').style="display:none;"
+            document.getElementById('showQuestion').style="display:none;"
+            document.getElementById('userAnswer').style="display:none;"
+            document.getElementById('acceptar').style="display:none;"
+            document.getElementsByClassName('letter').style="display:none;"
+            document.getElementById('Okrepeat').style="display:none;"//pq no es mostri
+            document.getElementById('repeatPasapalabra').style="display:none;"
+            document.getElementById('showresults').style="display:none;"
+            document.getElementById('results').style="display:none;"
+            document.getElementById("end").style="display:none;" 
+            document.getElementById("questionPlayAgain").style="display:block;"
+            document.getElementById("playAgainYes").style="display:block;"
+            document.getElementById("playAgainNo").style="display:block;"
+           
+        }
+    }, 1000);
+}
+
+
+
+window.onload = function () {
+    var twoMinutes = 3 * 1,
+        display = document.querySelector('#time');
+    startTimer(twoMinutes, display);
```

Comment:
You are calling `startTimer()` function in an `onload` event. This behavior start the timer a few seconds after to enter in the website.

I think you should call `startTimer()` function when you press the **start button**.

Reply:
thanks, I will try it!